### PR TITLE
Support parallel execution

### PR DESCRIPTION
**Pull request by notogawa**
and add a unit test (includes performance test).

### CODE REVIEW

File: videocore6/driver.py [hunk 1]
```diff
@@ -194,4 +206,4 @@ def execute(self, code, uniforms=None, timeout_sec=10):
         # XXX: Separate function
         for bo_handle in self.bo_handles:
             self.drm.v3d_wait_bo(bo_handle,
-                                 timeout_ns=int(timeout_sec / 1e-9))
+                                 timeout_ns=int(10 * timeout_sec / 1e-9))
```
Comment (Terminus-IMRC) [inline videocore6/driver.py]: What is the purpose of this? Regarding the result from `test_clock` in `tests/test_qpu.py`, I think this parameter is not needed to be changed.

Comment (notogawa) [reply videocore6/driver.py]: Right. After my performance experiment, I forgot to restore it.

File: videocore6/driver.py [hunk 2]
```diff
@@ -170,14 +170,26 @@ def program(self, prog, *args, **kwargs):
 
         return code
 
-    def execute(self, code, uniforms=None, timeout_sec=10):
+    def execute(self, code, uniforms=None, timeout_sec=10, workgroup=(16, 1, 1), wgs_per_sg=16, thread=1):
+
+        wg_x, wg_y, wg_z = workgroup
+        wg_size = wg_x * wg_y * wg_z
+        wgs_per_sg = wgs_per_sg
```
[the remark this answers is not in the document]
Comment (notogawa) [reply videocore6/driver.py]: Yes, thx.

**Comment by Terminus-IMRC** [comment]
Thank you for the fix. Your investigation is really nice!

**Pull request merged**